build: update golang.org/x/tools and fix some lint failures

RaduBerinde opened the pull request:
Update `golang.org/x/tools` and fix new `lintonbuild` failures
where we were passing an arbitrary string as a format string.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/129208)
<!-- Reviewable:end -->

RaduBerinde committed:
build: update golang.org/x/tools and fix some lint failures

Update `golang.org/x/tools` and fix new `lintonbuild` failures
where we were passing an arbitrary string as a format string.

Epic: none
Release note: None

srosenberg reviewed (approved):
Thanks!

RaduBerinde commented:
TFTR!

bors r+
